### PR TITLE
file upload issue in WYSIWYG editor on product page

### LINKED ISSUE #38026: [Issue] file upload issue in WYSIWYG editor on product page
This issue is automatically created based on existing pull request: magento/magento2#38025: file upload issue in WYSIWYG editor on product page

---------

### Title 
The folder-tree are not visible and getting error “We can't upload the file to the current folder right now. Please try another folder.” when we try to upload image 

### Description (*)

There is an issue encountered when attempting to upload an image to a folder within the WYSIWYG editor on a product page, specifically within the short description in content section. The folders are not visible, and as a result, we cannot upload the image and we receive an error message stating, 'We can't upload the file to the current folder right now. Please try another folder.' This problem only occurs when you expand the 'Image and Video' tab first. And getting the error in console  as shown in below image
 
image

### Manual testing scenarios (*)
1: Access the Magento admin panel and navigate to 'Catalog' -> 'Products.' and  Select any product.

2: in the product page, first, expand the 'Image and Videos' tab, as shown in the image below. 

image

3: Next, expand the 'Content' tab and in 'Short Description' section click on the 'Insert Image' button and select the 'Source' button in the WYSIWYG editor, as indicated in the image below

image

4: after clicking the 'Source' button, you may notice that the folders are not displayed, and as a result, you are unable to upload images within the 'Short Description.' as you can see in below image

image

### **Expected result **
After expanding image and videos tab first we should be able to see folder option and can insert image in short description

### **Actual result **
After expanding image and videos tab first we can not see folder option and can not insert image in short-description 

### **Aditional information**
In this issue, when we expand the 'Image and Videos' tab and then attempt to upload an image within the 'Short Description' content section, it triggers the 'tmp.init(el,options);'  in the lib/web/jquery/jstree/jquery.jstree.js file at line 123, as illustrated in the image below.

image

After the 'tmp.init(el, options)' function is triggered in the 'jquery.jstree.js' file, it call the init function in the 'tree-suggest.js' file. Please note that it call init function in  the 'tree-suggest.js' file  only when we initially expand the 'Image and Videos' tab first.

In the lib/web/mage/backend/tree-suggest.js file, it calls the ‘init’ function at line 58 however, an issue arises because the ‘init’ function does not receive the element. Consequently, ‘this.get_container’ becomes undefined 

This issue likely plays a significant role in the failure to display folders in the WYSIWYG editor when attempting to upload an image.the element contains the data of folder-tree so if it not assign correct we can not see folder tree when we try to insert image using  WYSIWYG editor in short-description

image
In the image provided above, you can see on line 77 and also on line 102, there is a call to the '_get_node' method. However, there isn't a method with this name in the file or jquery.jstree.js file. Instead, the correct method name is 'get_node.' Therefore, we need to make this correction to ensure the proper function is called.

### **Solution**

To resolve the issue, we need to make changes in the 'lib/web/mage/backend/tree-suggest.js' file. Specifically, we should pass the 'element' in the 'init' function at line 58. Afterward, we should call the 'init' function before 'get_container' and pass the 'data' (consisting of 'element' and 'option') to ensure that the 'init' function receives the required element. This modification ensures that the 'container' variable is not undefined. 

Additionally, we should replace instances of '_get_node' with 'get_node' on lines 77 and 102 to ensure that the correct method is called.

Once these modifications are implemented, the issue should be successfully resolved.you can see changes in below image

image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by DhruvBasan**
### Title 
The folder-tree are not visible and getting error “We can't upload the file to the current folder right now. Please try another folder.” when we try to upload image 

### Description (*)

There is an issue encountered when attempting to upload an image to a folder within the WYSIWYG editor on a product page, specifically within the short description in content section. The folders are not visible, and as a result, we cannot upload the image and we receive an error message stating, 'We can't upload the file to the current folder right now. Please try another folder.' This problem only occurs when you expand the 'Image and Video' tab first. And getting the error in console  as shown in below image
 
image

### Manual testing scenarios (*)
1: Access the Magento admin panel and navigate to 'Catalog' -> 'Products.' and  Select any product.

2: in the product page, first, expand the 'Image and Videos' tab, as shown in the image below. 

image

3: Next, expand the 'Content' tab and in 'Short Description' section click on the 'Insert Image' button and select the 'Source' button in the WYSIWYG editor, as indicated in the image below

image

4: after clicking the 'Source' button, you may notice that the folders are not displayed, and as a result, you are unable to upload images within the 'Short Description.' as you can see in below image

image

### **Expected result **
After expanding image and videos tab first we should be able to see folder option and can insert image in short description

### **Actual result **
After expanding image and videos tab first we can not see folder option and can not insert image in short-description 

### **Aditional information**
In this issue, when we expand the 'Image and Videos' tab and then attempt to upload an image within the 'Short Description' content section, it triggers the 'tmp.init(el,options);'  in the lib/web/jquery/jstree/jquery.jstree.js file at line 123, as illustrated in the image below.

image

After the 'tmp.init(el, options)' function is triggered in the 'jquery.jstree.js' file, it call the init function in the 'tree-suggest.js' file. Please note that it call init function in  the 'tree-suggest.js' file  only when we initially expand the 'Image and Videos' tab first.

In the lib/web/mage/backend/tree-suggest.js file, it calls the ‘init’ function at line 58 however, an issue arises because the ‘init’ function does not receive the element. Consequently, ‘this.get_container’ becomes undefined 

This issue likely plays a significant role in the failure to display folders in the WYSIWYG editor when attempting to upload an image.the element contains the data of folder-tree so if it not assign correct we can not see folder tree when we try to insert image using  WYSIWYG editor in short-description

image
In the image provided above, you can see on line 77 and also on line 102, there is a call to the '_get_node' method. However, there isn't a method with this name in the file or jquery.jstree.js file. Instead, the correct method name is 'get_node.' Therefore, we need to make this correction to ensure the proper function is called.

### **Solution**

To resolve the issue, we need to make changes in the 'lib/web/mage/backend/tree-suggest.js' file. Specifically, we should pass the 'element' in the 'init' function at line 58. Afterward, we should call the 'init' function before 'get_container' and pass the 'data' (consisting of 'element' and 'option') to ensure that the 'init' function receives the required element. This modification ensures that the 'container' variable is not undefined. 

Additionally, we should replace instances of '_get_node' with 'get_node' on lines 77 and 102 to ensure that the correct method is called.

Once these modifications are implemented, the issue should be successfully resolved.you can see changes in below image

image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#38026: file upload issue in WYSIWYG editor on product page

### CODE REVIEW

**Comment by DhruvBasan** [comment]
@magento run Unit Tests

**Comment by engcom-Bravo** [comment]
@magento create issue

**Comment by DhruvBasan** [comment]
@magento run all tests

File: lib/web/mage/backend/tree-suggest.js [hunk 1]
```diff
@@ -55,7 +55,8 @@ define([
         /**
          * @override
          */
-        init: function () {
+        init: function (el,options) {
+            init.call(this,el,options);
```
Comment (andrewbess) [inline lib/web/mage/backend/tree-suggest.js]: Let's add spaces between params

```suggestion
        init: function (el, options) {
            init.call(this, el, options);
```
Perhaps JS Coding Style requires it
https://javascript.info/coding-style

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @DhruvBasan 
Thank you for your contribution
Could you please check my suggestions and fix PR
Thank you in advance.

File: lib/web/mage/backend/tree-suggest.js (hunk 1 above)
Comment (DhruvBasan) [reply lib/web/mage/backend/tree-suggest.js]: Hello @andrewbess , I have implemented the suggestions you provided  on my pull request

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Integration Tests, WebAPI Tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @DhruvBasan,

Thanks for the collaboration!

The code changes seems good to us. But in accordance of [DOD](https://developer.adobe.com/commerce/contributor/guides/code-contributions/definition-of-done/#definition-of-done-1), please cover the changes with some automated test.

Thanks

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run WebAPI Tests,Functional Tests CE,Functional Tests B2B

**Comment by engcom-Echo** [comment]
Functional Tests CE,Functional Tests B2B failure are different on last two run on same commit.
WebAPI Tests are not related to PR changes.

Moving it to `Pending Review.`

**Review by andrewbess — APPROVED** [review]
Hello @DhruvBasan 
Thank you for your work here
Changes look nice for me
Approved

**Comment by engcom-Echo** [comment]
WebAPI Tests failure is known failure and not related to PR changes.
Functional Tests CE and Functional Tests B2B failure are different on last two run on same commit.
Functional Tests CE 
Run1
Screenshot 2024-02-08 at 4 08 53 PM

Run2
Screenshot 2024-02-08 at 4 09 11 PM

Functional Tests B2B 
Run1
Screenshot 2024-02-08 at 4 08 59 PM

Run2
Screenshot 2024-02-08 at 4 09 33 PM

Hence Moving it to `Merge In Progress`

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Static Tests

**Comment by engcom-Echo** [comment]
@magento run Static Tests,Integration Tests,Functional Tests CE,Functional Tests B2B

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run WebAPI Tests,Functional Tests B2B

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE,Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE ,Integration Tests

**Comment by engcom-Echo** [comment]
Functional Tests B2B failures are different on last two run on same commit.

Run1
Screenshot 2024-03-05 at 12 44 11 PM

Run2
Screenshot 2024-03-05 at 12 44 17 PM

**Comment by hostep** [comment]
Looks like this was merged in 2.4-develop recently in https://github.com/magento/magento2/commit/c6bc8300e771e5e96ce9ac6c116746fe3de80544

However, PR isn't marked as merged, probably because of the last 2 merge commits part of this PR.
So I guess this PR can be closed?

**Comment by engcom-Echo** [comment]
Closing this PR as it is merged to 2.4-develop. Commit link can be found [here](https://github.com/magento/magento2/pull/38025#issuecomment-2116319878)

**Pull request closed without merging**